lib/strategy/FragmentationStrategyShape.ts

```diff
@@ -0,0 +1,292 @@
+
+  /**
+   *
+   * @param {string} shapeDirectory - the directory of the shape
+   * @param {string} relativePath - the relatif path of the IRI
+   * @param {boolean|undefined} tripleShapeTreeLocator - indicate if a shape locator triple is generated
+   * @param {number|undefined} generationProbability - indicate if there is a probability [1, 100]
+   * to generate shape information on a specific resource in a container @range {float}
+   */
+  public constructor(shapeDirectory: string,
+    relativePath?: string,
+    tripleShapeTreeLocator?: boolean,
+    generationProbability?: number) {
+    super();
+    this.tripleShapeTreeLocator = tripleShapeTreeLocator;
+    this.relativePath = relativePath;
+    this.shapeMap = this.generateShapeMap(shapeDirectory);
+    this.generationProbability = generationProbability;
+    if (this.generationProbability !== undefined &&
+      (this.generationProbability > 100 || this.generationProbability < 0)) {
+      throw new Error('The probability to generate shape information should be between 0 and 100');
+    }
+  }
+
+  /**
+   * Generate a map of the resource type and the shape information
+   * @param {string} shapeFolder - directory containing the shape information
+   * @returns {Map<string, IShapeEntry>} a map of the resource with their shape information
+   */
+  private generateShapeMap(shapeFolder: string): Map<string, IShapeEntry> {
+    const shapeMap: Map<string, IShapeEntry> = new Map();
+    const config = JSON.parse(readFileSync(join(shapeFolder, 'config.json')).toString());
+    const shapes = config.shapes;
+    for (const [ dataType, shapeEntry ] of Object.entries(shapes)) {
+      shapeMap.set(dataType, {
+        shape: readFileSync(join(shapeFolder, (<IShapeEntry>shapeEntry).shape)).toString(),
+        directory: (<IShapeEntry>shapeEntry).directory,
+      });
+    }
+    return shapeMap;
+  }
+
+  /**
+   * Determine if the shape information should be generated.
+   * returns always true if the probability is not defined.
+   * @returns {boolean} indicate if the shape information should be generated
+   */
+  public shouldGenerate(): boolean {
+    return this.generationProbability === undefined ?
+      true :
+      Math.random() * 100 <= this.generationProbability;
+  }
+
+  /**
+   * From a quad generate the IRI based on the {@link FragmentationStrategySubject}.
+   * Evaluate if shape index information should be generated based on which quad has been handled and
+   * the shape config file provides. Generate the right iri for the shape indexes.
+   * The generation of the triples and the quad sink pushing is handled by other functions called by
+   * handleQuad.
+   * @param {RDF.Quad} quad - a quad
+   * @param {IQuadSink} quadSink - a quad sink
+   */
+  protected async handleQuad(quad: RDF.Quad, quadSink: IQuadSink): Promise<void> {
+    const iri = FragmentationStrategySubject.generateIri(quad, this.relativePath);
```

Comment:
Do we only need to do the following if `quad.subject` is a namednode?

Reply:
I want to be sure that we generate has the subject strategy, so I decided to use the same method.

Reply:
Sure, but I mean, do we need to wrap this in an if-statement, just like `FragmentationStrategySubject`?